dev/com.ibm.ws.wsoc/src/com/ibm/ws/wsoc/HandshakeProcessor.java

```diff
@@ -475,4 +479,33 @@ public static List<Extension> parseClientExtensions(List<String> list) {
         }
         return extensions;
     }
+
+    private URI buildFullURI(HttpServletRequest req) throws Exception {
+            StringBuilder builder = new StringBuilder();
+
+            String url = req.getRequestURL().toString();
+            String https = "https";
+            String http = "http";
+
+            if(url.startsWith(https)){
+                url = "wss" + url.substring(https.length(), url.length());
+            } else if(url.startsWith(http)){
+                url = "ws" + url.substring(http.length(), url.length());
+            }
+
+            if( !(url.startsWith("ws:") || url.startsWith("wss:")) ){
+                throw new Exception("Scheme is not of type ws or wss.");
```

Comment:
@pnicolucci  Let me know if I should do something else there.

Reply:
If this is a user-visible exception it should be translated, we can do that in a follow on PR if you open up an issue so we can address it after we merge this one.

Reply:
@volosied  looks like you have already created a message! So we can resolve this one.

Reply:
Ye